Fix test flake (signing & verification precompile intermittent error)

##### SUMMARY
<!--- Explain the problem briefly below -->

https://github.com/oasisprotocol/sapphire-paratime/actions/runs/6845602636/job/18611054478


```
  1) Signing
       Ed25519 (Oasis):

      AssertionError: expected true to equal false
      + expected - actual

      -true
      +false
      
      at testSignThenVerify (test/signing.ts:35:7)
      at processTicksAndRejections (node:internal/process/task_queues:95:5)
      at async Context.<anonymous> (test/signing.ts:131:9)
```

The test verifies that if the message of a signature is changed the verification must fail, however sometimes it returns true. This is very concerning and we need to find how to reproduce this error reliably.

At the moment it happens on maybe 1/50 CI runs and is intermittent - it's not specific to a single signing algorithm.

--- a/contracts/test/signing.ts
+++ b/contracts/test/signing.ts
@@ -7,6 +7,13 @@ import { ethers } from 'hardhat';
 import { SigningTests__factory } from '../typechain-types/factories/contracts/tests';
 import { SigningTests } from '../typechain-types/contracts/tests/SigningTests';
 
+function randomBytesUnlike(len: number, orig: Buffer): Buffer {
+  do {
+    var bytes = randomBytes(len);
+  } while (len > 0 && bytes.equals(orig));
+  return bytes;
+}
+
 async function testSignThenVerify(
   se: SigningTests,
   alg: number,
@@ -29,7 +36,7 @@ async function testSignThenVerify(
         alg,
         keypair.publicKey,
         ctx,
-        randomBytes(msg.length),
+        randomBytesUnlike(msg.length, msg),
         sig,
       ),
     ).equal(false);
@@ -41,7 +48,7 @@ async function testSignThenVerify(
       await se.testVerify(
         alg,
         keypair.publicKey,
-        randomBytes(ctx.length),
+        randomBytesUnlike(ctx.length, ctx),
         msg,
         sig,
       ),

